opensrp-reveal/src/main/java/org/smartregister/reveal/application/RevealApplication.java

```diff
@@ -176,4 +190,35 @@ public LocationRepository getLocationRepository() {
         }
         return locationRepository;
     }
+
+    public AllSettings getSettingsRepository() {
+        return getInstance().getContext().allSettings();
+    }
+
+    public void processGlobalConfigs() {
+        Setting globalSettings =  getSettingsRepository().getSetting(GLOBAL_CONFIGS);
+        populateGlobalConfigs(globalSettings);
+    }
+
+    private void populateGlobalConfigs(@NonNull Setting setting) {
+        if (setting == null) {
+            return;
+        }
+        try {
+            JSONArray settingsArray = new JSONArray(setting.getValue());
+            for (int i = 0; i < settingsArray.length(); i++) {
+                JSONObject jsonObject = settingsArray.getJSONObject(i);
+                String value = jsonObject.optString(VALUE, null);
+                if (value != null) {
+                    globalConfigs.put(jsonObject.optString(KEY), value);
```

Comment:
Will this you have null keys in the map, why not check if key is not null first

Reply:
Done.